frontend/src/editor/speaker_colors.tsx

```diff
@@ -1,5 +1,6 @@
-import React, { useEffect, useState } from 'react';
+import React, { useMemo } from 'react';
 import { Editor } from 'slate';
+import { useSpeakerIDs } from '../utils/document';
 
 export const SpeakerColorsContext = React.createContext<Record<string, string>>({});
```

Comment:
I think at this point we could just remove this whole context hack and move the remaining code into `utils/document.ts`

Reply:
Also we might want to have a way to have memoized selectors onto the document at some point. maybe not now but lets create an issue?

Reply:
Well, the memoized selectors onto the document seem a lot like what context is doing to me?
I am not that fluent in react, is context not how you usually do thing?

Reply:
Yes but without the need to express it explicitly. for the time being I think we are good with just re-calculating it sometimes as it is neither very costly nor very often done